### PR TITLE
add Travis CI

### PR DESCRIPTION
**Pull request by 1ace**

### CODE REVIEW

**Comment by djdeath** [comment]
Thanks a lot for doing this!

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+language: c
+
+cache:
+  - ccache
+
+os: linux
+dist: bionic
```
Comment (djdeath) [inline .travis.yml]: Checking the Vulkan-Loader, it seems to use xenial. I wonder if we should align. Or is it causing issues?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+language: c
```
Comment (djdeath) [inline .travis.yml]: Maybe cpp? It's actually both, but I don't know if it matters :)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+cache:
+  - ccache
+
+os: linux
```
Comment (djdeath) [inline .travis.yml]: We definitively want to test windows etc...

Here is what Vulkan-Loader has : 

```
matrix:
  # Show final status immediately if a test fails.                                                
  fast_finish: true
  allow_failures:
    - env: CHECK_COMMIT_FORMAT=ON
    - env: VULKAN_BUILD_TARGET=GN
  include:
    # Linux GCC debug build.                                                                      
    - os: linux
      compiler: gcc
      env: VULKAN_BUILD_TARGET=LINUX
    # Linux clang debug build.                                                                    
    - os: linux
      compiler: clang
      env: VULKAN_BUILD_TARGET=LINUX
    # Linux GN debug build                                                                        
    - os: linux
      env: VULKAN_BUILD_TARGET=GN
    # MacOS clang debug build.                                                                    
    - os: osx
      compiler: clang
      env: VULKAN_BUILD_TARGET=MACOS
    # Check for proper clang formatting in the pull request.                                      
    - env: CHECK_FORMAT=ON
    # Check for proper commit message formatting for commits in PR                                
    - env: CHECK_COMMIT_FORMAT=ON
```

File: .travis.yml (hunk 1 above)
Comment (1ace) [reply .travis.yml]: That's actually a leftover from earlier versions.

I kinda want to just drop these two lines and use the Travis defaults; is that ok with you?

**Comment by djdeath** [comment]
More general question : how do you test this stuff?

File: .travis.yml (hunk 2 above)
Comment (1ace) [reply .travis.yml]: I think that's used to pick the tools installed by default. It's a single value, but `cpp` probably includes all the C tools too so I'll set it to that :+1:

File: .travis.yml (hunk 3 above)
Comment (1ace) [reply .travis.yml]: Ah, I didn't think to look at what other KHR projects are doing, I just wrote it from scratch :upside_down_face:

I'll rewrite it to match the Loader travis config :+1:

**Comment by 1ace** [comment]
> how do you test this stuff?

If you mean "how to test changes to the CI", you just open https://travis-ci.org/dj-death, look for "Vulkan-ExtensionLayer" and enable it. When you push your change to a branch on your fork it will automatically run and you'll get an email if it fails, or you can just open https://travis-ci.org/dj-death/Vulkan-ExtensionLayer

File: .travis.yml (hunk 1 above)
Comment (djdeath) [reply .travis.yml]: I think it would make more sense to just use a similar setup to the rest of the Khronos repos.

File: .travis.yml (hunk 3 above)
Comment (djdeath) [reply .travis.yml]: Yeah, I think we should probably add the following files : 
```
scripts/update_deps.py
scripts/known_good.json
```
So the right headers are downloaded.

**Comment by djdeath** [comment]
> > how do you test this stuff?
> 
> If you mean "how to test changes to the CI", you just open https://travis-ci.org/dj-death, look for "Vulkan-ExtensionLayer" and enable it. When you push your change to a branch on your fork it will automatically run and you'll get an email if it fails, or you can just open https://travis-ci.org/dj-death/Vulkan-ExtensionLayer

:+1 Thanks!

**Comment by 1ace** [comment]
OK, rewrote to test:
- linux {32,64}x{gcc,clang}
- macos with clang
- windows (default compiler, I'm guessing that's msvc)

The 32-bit builds are failing because of what #9 fixes.

The MacOS and Windows builds are failing because the code assumes too much and needs to be fixed.

Do we want to merge the CI as is, or leaving it in this PR until it's all green?
I think merging now is better as it allows easy testing of MacOS/Window PRs

**Comment by djdeath** [comment]
> Do we want to merge the CI as is, or leaving it in this PR until it's all green?
> I think merging now is better as it allows easy testing of MacOS/Window PRs

Yeah, I think it makes to make before getting MacOS & Windows. Maybe we can land the other PR first to have at least a clean Linux build?

**Comment by djdeath** [comment]
Or maybe even better, it's comment out the systems that don't currently work and I'll add a PR for each that uncomments the build together with the fix.

**Comment by 1ace** [comment]
> Maybe we can land the other PR first to have at least a clean Linux build?

I think that's best :+1:

> Or maybe even better, it's comment out the systems that don't currently work and I'll add a PR for each that uncomments the build together with the fix.

I'll comment out the MacOS & Windows builds :+1:

**Pull request merged**